### PR TITLE
docs: blog post on AI Lab recipes

### PR DESCRIPTION
**Pull request by feloy**
### What issues does this PR fix or reference?

Fixes https://github.com/containers/podman-desktop-internal/issues/342

### How to test this PR?

- [ ] Tests are covering the bug fix or the new feature

### CODE REVIEW

File: website/blog/2024-11-12-ai-lab-first-app.md [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+
+```
+Give me a list of pages in the website www.podman-desktop.io related to "build an image" as JSON output as an array of objects with 2 fields name and url
+```
+
+This time, we have received a response in JSON format, which is more suitable for our needs.
+
+![a prompt with structured output](./img/ai-lab-first-app/05-ai-lab-demo-prompt-json.png)
+
+We don't ecxpect the user to ask such a precise question, and we would prefer to send to the model the exact question of the user, without modifying it in real time. For this purpose, chat models provide a system prompt feature. The system prompt can be defined at the beginning of the chat session.
```
Comment (axel7083) [inline website/blog/2024-11-12-ai-lab-first-app.md]:
```suggestion
We don't expect the user to ask such a precise question, and we would prefer to send to the model the exact question of the user, without modifying it in real time. For this purpose, chat models provide a system prompt feature. The system prompt can be defined at the beginning of the chat session.
```

File: website/blog/2024-11-12-ai-lab-first-app.md [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+
+![a session with a system prompt](./img/ai-lab-first-app/06-ai-lab-demo-system-prompt.png)
+
+> Please note that this section was not a course on writing the best prompt, I'm sure you will find much more efficient prompts for this purpose. The purpose of this section is to demonstrate how you can iterate with Podman AI Lab to refine the prompts you want to use for your application.
+
+## Testing a recipe
+
+Now that we have a suitable prompt to use for our application, it is time to start our application itself. 
+
+Many developers prefer to have a working example of application to start with, and Podman AI Lab provides such examples with a catalog of recipes, visible in the page `AI Apps > Recipes Catalog`.
```
Comment (feloy) [inline website/blog/2024-11-12-ai-lab-first-app.md]:
```suggestion
Many developers prefer to have a working example of application to start with, and Podman AI Lab provides such examples with a catalog of recipes, visible in the page `AI Apps > Recipe Catalog`.
```

File: website/blog/2024-11-12-ai-lab-first-app.md [hunk 3]
```diff
@@ -0,0 +1,218 @@
...
+        - arm64
+        - amd64
+      ports:
+        - 8501
+      image: quay.io/ai-lab/chatbot:latest
+```
+
+The file defines two containers, one for the inference server and one for the application itself.
+
+The first container, for the inference server, is generic and can be reused for any app by using a chat model.
```
Comment (feloy) [inline website/blog/2024-11-12-ai-lab-first-app.md]:
```suggestion
The first container, for the inference server, is generic and can be reused for any app using a chat model.
```

File: website/blog/2024-11-12-ai-lab-first-app.md [hunk 4]
```diff
@@ -0,0 +1,218 @@
...
+      "recommended": [
+        "hf.TheBloke.mistral-7b-instruct-v0.2.Q4_K_M"
+      ],
+      "backend": "llama-cpp"
+    }
+  ]
+}
+```
+
+By adding thie above content to `$HOME/.local/share/containers/podman-desktop/extensions-storage/redhat.ai-lab/user-catalog.json`, you should now be able to see a new recipe `Search Podman-desktop.io` in the recipes catalog of Podman AI Lab, and run it as any other recipe. And, of course, you can share this file with your colleague to share with them your latest experiment.
```
Comment (feloy) [inline website/blog/2024-11-12-ai-lab-first-app.md]:
```suggestion
By adding thie above content to `$HOME/.local/share/containers/podman-desktop/extensions-storage/redhat.ai-lab/user-catalog.json`, you should now be able to see a new recipe `Search Podman-desktop.io` in the recipe catalog of Podman AI Lab, and run it as any other recipe. And, of course, you can share this file with your colleagues to share with them your latest experiment.
```

File: website/blog/2024-11-12-ai-lab-first-app.md [hunk 5]
```diff
@@ -0,0 +1,218 @@
...
+hide_table_of_contents: false
+---
+
+Red Hat provides an extension to Podman Desktop, Podman AI Lab, which lets developers discover examples of applications by using large language models (LLMs), and gives them a framework to create their own AI-based applications and share them with their team.
+
+We will discover, through this article, the different steps to create our first AI application, and to add it to the catalog of recipes of Podman AI Lab.
+
+# Our first AI Application
+
+For our first experiment, we will work on a micro-service for the podman-desktop.io website. The micro-service would receive the search terms from the website, and would ask the model to find the best matching pages, before to return the result to the website.
```
Comment (Firewall) [inline website/blog/2024-11-12-ai-lab-first-app.md]:
```suggestion
For our first experiment, we will work on a micro-service for the podman-desktop.io website. The micro-service would receive the search terms from the website, and would ask the model to find the best matching pages, before returning the result to the website.
```

File: website/blog/2024-11-12-ai-lab-first-app.md [hunk 6]
```diff
@@ -0,0 +1,218 @@
...
+
+![my first app](./img/ai-lab-first-app/00-my-first-ai-app.png)
+
+## Preparing Podman Desktop and Podman AI Lab
+
+If you haven't done it yet, the first step is to [install Podman Desktop and its extension Podman AI Lab](https://podman-desktop.io/docs/ai-lab/installing).
```
Comment (Firewall) [inline website/blog/2024-11-12-ai-lab-first-app.md]:
```suggestion
If you haven't done it yet, first [install Podman Desktop and its extension Podman AI Lab](https://podman-desktop.io/docs/ai-lab/installing).
```

File: website/blog/2024-11-12-ai-lab-first-app.md [hunk 7]
```diff
@@ -0,0 +1,218 @@
...
+
+![GPU support for inference servers preference is enabled](./img/ai-lab-first-app/02-ai-lab-demo-gpu-preference.png)
+
+## Testing a prompt with a model
+
+Podman AI Lab provides a catalog of open source models that can be used locally. You can go to the `Models > Catalog` page to download the model of your choice. For this article, we will use the `Mistral-7B-instruct` model.
+
+![Mistral model is downloaded](./img/ai-lab-first-app/03-ai-lab-demo-mistral-model-downloaded.png)
+
+Once a model is downloaded, we can test and interact with this model, to try to find the best prompt for our application. For chat models, Podman AI Lab provides a `Playground`, so we can test different prompts and validate that the responses of the model are adequate.
```
Comment (Firewall) [inline website/blog/2024-11-12-ai-lab-first-app.md]:
```suggestion
Once a model is downloaded, we can test and interact with this model to try to find the best prompt for our application. For chat models, Podman AI Lab provides a `Playground`, so we can test different prompts and validate that the responses of the model are adequate.
```

File: website/blog/2024-11-12-ai-lab-first-app.md [hunk 8]
```diff
@@ -0,0 +1,218 @@
...
+
+Let's start a new playground (from the `Models > Playgrounds` menu), and send our first prompt:
+
+```
+Give me a list of pages in the website www.podman-desktop.io related to "build an image"
+```
+
+The model should reply with some list of pages, in a human-readable form (see the screenshot below, for a response we have received).
```
Comment (Firewall) [inline website/blog/2024-11-12-ai-lab-first-app.md]:
```suggestion
The model should reply with some list of pages, in a human-readable form (see the screenshot below, for the response we received).
```

File: website/blog/2024-11-12-ai-lab-first-app.md [hunk 9]
```diff
@@ -0,0 +1,218 @@
...
+
+![a first prompt with human readable output](./img/ai-lab-first-app/04-ai-lab-demo-prompt-1.png)
```
Comment (Firewall) [inline website/blog/2024-11-12-ai-lab-first-app.md]:
```suggestion
![a first prompt with human-readable output](./img/ai-lab-first-app/04-ai-lab-demo-prompt-1.png)
```

File: website/blog/2024-11-12-ai-lab-first-app.md [hunk 10]
```diff
@@ -0,0 +1,218 @@
...
+
+The problem is that the response is in human readable form, but we don't want the API to return this response as is, we want to have the name and the url of the pages, and send them to the website, so the website can display these pages with its preferred format.
```
Comment (Firewall) [inline website/blog/2024-11-12-ai-lab-first-app.md]:
```suggestion
The problem is that the response is in human-readable form, but we don't want the API to return this response as is. We want to have the name and the url of the pages, and send them to the website, so the website can display these pages with its preferred format. 
```

File: website/blog/2024-11-12-ai-lab-first-app.md [hunk 11]
```diff
@@ -0,0 +1,218 @@
...
+
+The problem is that the response is in human readable form, but we don't want the API to return this response as is, we want to have the name and the url of the pages, and send them to the website, so the website can display these pages with its preferred format. 
+
+For this, we can try to ask the model to reply with a structured response, with the following prompt:
+
+```
+Give me a list of pages in the website www.podman-desktop.io related to "build an image" as JSON output as an array of objects with 2 fields name and url
+```
+
+This time, we have received a response in JSON format, which is more suitable for our needs.
```
Comment (Firewall) [inline website/blog/2024-11-12-ai-lab-first-app.md]:
```suggestion
This time, we received a response in JSON format, which is more suitable for our needs.
```

File: website/blog/2024-11-12-ai-lab-first-app.md [hunk 12]
```diff
@@ -0,0 +1,218 @@
...
+
+![a prompt with structured output](./img/ai-lab-first-app/05-ai-lab-demo-prompt-json.png)
+
+We don't expect the user to ask such a precise question, and we would prefer to send to the model the exact question of the user, without modifying it in real time. For this purpose, chat models provide a system prompt feature. The system prompt can be defined at the beginning of the chat session.
```
Comment (Firewall) [inline website/blog/2024-11-12-ai-lab-first-app.md]:
```suggestion
We don't expect the user to ask such a precise question, and we would prefer to send to the model the exact question of the user, without modifying it in real time. To achieve this, chat models provide a system prompt feature. The system prompt can be defined at the beginning of the chat session.
```

**Review by Firewall — CHANGES_REQUESTED** [review]

**Comment by feloy** [comment]
thanks @Firewall for your suggestions/corrections. I've applied them.

**Review by Firewall — APPROVED** [review]

File: website/blog/2024-11-19-ai-lab-first-app.md [hunk 13]
```diff
@@ -0,0 +1,220 @@
...
+Podman AI Lab provides a catalog of open source models that can be used locally. You can go to the `Models > Catalog` page to download the model of your choice. For this article, we will use the `Mistral-7B-instruct` model.
+
+![Mistral model is downloaded](./img/ai-lab-first-app/03-ai-lab-demo-mistral-model-downloaded.png)
+
+Once a model is downloaded, we can test and interact with this model to try to find the best prompt for our application. For chat models, Podman AI Lab provides a `Playground`, so we can test different prompts and validate that the responses of the model are adequate.
+
+Let's start a new playground (from the `Models > Playgrounds` menu), and send our first prompt:
+
+```
+Give me a list of pages in the website www.podman-desktop.io related to "build an image"
```
Comment (jeffmaury) [inline website/blog/2024-11-19-ai-lab-first-app.md]:
```suggestion
Give me a list of pages in the website podman-desktop.io related to "build an image"
```

File: website/blog/2024-11-19-ai-lab-first-app.md [hunk 14]
```diff
@@ -0,0 +1,220 @@
...
+The model should reply with some list of pages, in a human-readable form (see the screenshot below, for the response we received).
+
+![a first prompt with human-readable output](./img/ai-lab-first-app/04-ai-lab-demo-prompt-1.png)
+
+The problem is that the response is in human-readable form, but we don't want the API to return this response as is. We want to have the name and the url of the pages, and send them to the website, so the website can display these pages with its preferred format. 
+
+For this, we can try to ask the model to reply with a structured response, with the following prompt:
+
+```
+Give me a list of pages in the website www.podman-desktop.io related to "build an image" as JSON output as an array of objects with 2 fields name and url
```
Comment (jeffmaury) [inline website/blog/2024-11-19-ai-lab-first-app.md]:
```suggestion
Give me a list of pages in the website podman-desktop.io related to "build an image" as JSON output as an array of objects with 2 fields name and url
```

File: website/blog/2024-11-19-ai-lab-first-app.md [hunk 15]
```diff
@@ -0,0 +1,220 @@
...
+This time, we received a response in JSON format, which is more suitable for our needs.
+
+![a prompt with structured output](./img/ai-lab-first-app/05-ai-lab-demo-prompt-json.png)
+
+We don't expect the user to ask such a precise question, and we would prefer to send to the model the exact question of the user, without modifying it in real time. To achieve this, chat models provide a system prompt feature. The system prompt can be defined at the beginning of the chat session.
+
+Podman AI Lab supports this feature, let's restart a Playground session with the following system prompt:
+
+```
+Give me a list of pages in the website www.podman-desktop.io related to the request as JSON output as an array of objects with 2 fields name and url
```
Comment (jeffmaury) [inline website/blog/2024-11-19-ai-lab-first-app.md]:
```suggestion
Give me a list of pages in the website podman-desktop.io related to the request as JSON output as an array of objects with 2 fields name and url
```

File: website/blog/2024-11-19-ai-lab-first-app.md [hunk 16]
```diff
@@ -0,0 +1,220 @@
...
+
+> Please note that this section was not a course on writing the best prompt, I'm sure you will find much more efficient prompts for this purpose. The purpose of this section is to demonstrate how you can iterate with Podman AI Lab to refine the prompts you want to use for your application.
+
+## Testing a recipe
+
+Now that we have a suitable prompt to use for our application, it is time to start our application itself. 
+
+Many developers prefer to have a working example of application to start with, and Podman AI Lab provides such examples with a catalog of recipes, visible in the page `AI Apps > Recipe Catalog`.
+
+Let's select the Chatbot recipe (`More details` link on the Chatbot card), and start it with the Mistral model (by pressing the `Start` button and filling the form).
```
Comment (jeffmaury) [inline website/blog/2024-11-19-ai-lab-first-app.md]: Any reason not to use a Granite model here ?

File: website/blog/2024-11-19-ai-lab-first-app.md [hunk 17]
```diff
@@ -0,0 +1,220 @@
...
+    MessagesPlaceholder(variable_name="history"),
+    ("user", "{input}")
+])
+```
+
+This is exactly what we want to do in our application, we will be able to indicate here the system prompt we have found earlier.
+
+## Creating our own app
+
+Adapting the source code for the purpose of our application is out of the scope of this article, let's see the result in [our app repository](https://github.com/feloy/podman-ai-lab-demo/blob/main/app/service.py).
```
Comment (jeffmaury) [inline website/blog/2024-11-19-ai-lab-first-app.md]: User owned repo, I'm not a big fan, can't we have it as a sub folder of this PR ?

File: website/blog/2024-11-19-ai-lab-first-app.md [hunk 18]
```diff
@@ -0,0 +1,220 @@
...
+      "recommended": [
+        "hf.TheBloke.mistral-7b-instruct-v0.2.Q4_K_M"
+      ],
+      "backend": "llama-cpp"
+    }
+  ]
+}
+```
+
+By adding thie above content to `$HOME/.local/share/containers/podman-desktop/extensions-storage/redhat.ai-lab/user-catalog.json`, you should now be able to see a new recipe `Search Podman-desktop.io` in the recipe catalog of Podman AI Lab, and run it as any other recipe. And, of course, you can share this file with your colleagues to share with them your latest experiment.
```
Comment (jeffmaury) [inline website/blog/2024-11-19-ai-lab-first-app.md]: The file is the whole content of the user catalog so it should replace the file and not be appended

File: website/blog/2024-11-19-ai-lab-first-app.md (hunk 17 above)
Comment (feloy) [reply website/blog/2024-11-19-ai-lab-first-app.md]: The problem had also been raised in the PR https://github.com/redhat-developer/podman-desktop-demo/pull/58

Is it ok if we reference from this blog post the sources in `https://github.com/redhat-developer/podman-desktop-demo`? If it is ok, I'll update the PR above to include the sources of the repo in `https://github.com/redhat-developer/podman-desktop-demo`

File: website/blog/2024-11-19-ai-lab-first-app.md (hunk 16 above)
Comment (feloy) [reply website/blog/2024-11-19-ai-lab-first-app.md]: I tested with granite but it struggles responding in JSON format

File: website/blog/2024-11-19-ai-lab-first-app.md (hunk 17 above)
Comment (feloy) [reply website/blog/2024-11-19-ai-lab-first-app.md]: Note that if we place the sources in a repo, fetching the recipe from AI Lab will fetch the whole repository. We need to take care of placing it in a reasonably small one

Comment (slemeur) [reply website/blog/2024-11-19-ai-lab-first-app.md]: Yeah, that's ok to point to the repository on https://github.com/redhat-developer/podman-desktop-demo

Comment (jeffmaury) [reply website/blog/2024-11-19-ai-lab-first-app.md]: > Note that if we place the sources in a repo, fetching the recipe from AI Lab will fetch the whole repository. We need to take care of placing it in a reasonably small one

Came to my mind after I posted the comment

File: website/blog/2024-11-26-ai-lab-first-app.md [hunk 19]
```diff
@@ -0,0 +1,220 @@
...
+        with a list of pages in the website https://www.podman-desktop.io related to my query
+    """),
+    MessagesPlaceholder(variable_name="history"),
+    ("user", "{input}")
+])
+```
+
+## Testing my own app locally
+
+To iterate during the development of aour app, we can test our app locally in our host system, while using the model served by Podman AI Lab. For this, we need to start a new model service from the page `Models > Services`, by clicking the `New Model Service`, then choosing the appropriate model (`Mistral-7B-instruct` in our case), and specifying a port number (let's say 56625).
```
Comment (jeffmaury) [inline website/blog/2024-11-26-ai-lab-first-app.md]:
```suggestion
To iterate during the development of our app, we can test our app locally in our host system, while using the model served by Podman AI Lab. For this, we need to start a new model service from the page `Models > Services`, by clicking the `New Model Service`, then choosing the appropriate model (`Mistral-7B-instruct` in our case), and specifying a port number (let's say 56625).
```

**Review by jeffmaury — APPROVED** [review]
LGTM

**Pull request merged**